lib/timecop/time_extensions.rb

```diff
@@ -48,7 +48,17 @@ def strptime_with_mock_date(str = '-4712-01-01', fmt = '%F', start = Date::ITALY
           "supports Date::ITALY for the start argument."
       end
 
-      Time.strptime(str, fmt).to_date
+      d = Date._strptime(str, fmt) || Date.strptime_without_mock_date(str, fmt)
```

Comment:
Can you please explain why is there are `_strptime` or `strptime_without_mock_date`, and not just `strptime_without_mock_date` used?

Reply:
Based on the docs https://ruby-doc.org/stdlib-2.6/libdoc/date/rdoc/Date.html#method-c-_strptime returns a hash instead of a date object. I think I was thinking I didn't need the actual date object, but I wrote that too long ago to remember.

It looks like `strptime` actually calls `_strptime` internally, so we could probably just go with `strptime_without_mock_date`. Do the tests still pass that way?

cc https://github.com/travisjeffery/timecop/pull/242 <--- seems better to merge